writing/paper.md

```diff
@@ -2,84 +2,243 @@
+
+1. Dataset: From the provided URL, a dataset is obtained that will be used to train and test the spam message classifier. The labeled messages in the dataset are represented by the "v1" column, which stands for labels( spam or non-spamm), and the 'v2' column for text messages.
 
-TODO: This section describes your experimental set up and evaluation. It should also produce and describe the results of your study. The subsection titles below offer a typical structure used for this section.
+2. Data loading and processing: The requests library is used to load the dataset into a Pandas DataFrame. To make sure the dataset is loaded correctly, the column names are checked.
 
-#### Experimental Design
+3. Data splitting: Using the Sklearn's train _ test _ split function, the dataset is divided into training and testing sets. module for model selection. A random state of 42 is used for reproducibility, and the testing set size is set at 20% of the total data.
+
+4. The CountVectorizer from sklearn is a feature extraction. extraction of features. The text data is transformed into numerical features using the text module. The training and testing data are both transformed using the vectorizer, which is compatible with the training data.
+
+5. Model training: The sklearn's MultinomialNB classifier. The transformed training data is used to train the naive _ bayes module. Due to its efficiency in text classification tasks, this classifier was chosen.
+
+6. Performance evaluation: Using the trained classifier, the testing data are predicted. The sklearn's corresponding functions are used to calculate the accuracy, precision, recall, and F1 score, moduleS for metrics. These metrics reflect the classifier's ability to accurately classify spam and non-spam messages as well as its overall performance.
+
+The evaluation of the spam message classifier using the provided code yielded the following results:
+
+Accuracy: 0.9838565022421525
+
+recision: 0.9852941176470589
+
+Recall: 0.8933333333333333
+
+F1 score: 0.9370629370629371
+
+These metrics show that the classifier effectively categorizes both spam and non-spam messages. The classifier's high accuracy, precision, recall, and F1 score values imply that it can tell spam messages apart from non-spam messages.
+It's crucial to remember that the type of dataset used and the characteristics of the messages can have an impact on how well the classifier performs. To evaluate the classifier's effectiveness on various datasets and look into potential enhancements or optimizations, additional experimentation and evaluation are possible.
+
+![Screenshot 2023-07-27 at 1 04 46 PM 2](https://github.com/LS102-summer2023/project-mishikakansal/assets/136472000/0f2a9615-9003-4de4-b777-15ae3530bcd6)     (22)
 
-TODO: Especially as it pertains to responsible computing, if conducting experiments or evaluations that involve particular ethical considerations, detail those issues here.
 
 #### Evaluation
```

Comment:
This section would have been more effective if it was written as a cohesive text and not a list of results, and provided a detailed analysis of results, backed up by citations or conducted experiments.